thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/MetricConfigResource.java

```diff
@@ -160,4 +159,23 @@ public String viewMetricConfig(@NotNull @QueryParam("dataset") String dataset, @
     return rootNode.toString();
   }
 
+  @GET
+  @Path("/view")
+  @Produces(MediaType.APPLICATION_JSON)
+  public MetricConfigDTO viewMetricConfigByIdOrName(@QueryParam("id") String id,
+      @QueryParam("dataset") String dataset,
+      @QueryParam("metric") String metric) {
+
+    MetricConfigDTO metricConfigDTO = null;
+    if (StringUtils.isBlank(id) && (StringUtils.isBlank(dataset) || StringUtils.isBlank(metric))) {
```

Comment:
This line might not work as what you want it to do because the default value for a String is "undefined" if user's value is not given.